back/engines/commercial/multi_tenancy/app/models/tenant.rb

```diff
@@ -76,53 +70,24 @@ def self.current
   end
 
   def self.settings(*path)
-    ActiveSupport::Deprecation.warn('Tenant::settings is deprecated. Use AppConfiguration::settings instead.')
+    ErrorReporter.report_msg('Tenant::settings is deprecated. Use AppConfiguration#settings instead.')
     AppConfiguration.instance.settings(*path)
   end
 
-  def self.settings_json_schema_str
-    ActiveSupport::Deprecation.warn('Tenant::settings_json_schema_str is deprecated. Use AppConfiguration::Settings.json_schema_str_str instead.')
-    AppConfiguration::Settings.json_schema_str
-  end
-
-  def self.settings_json_schema
-    ActiveSupport::Deprecation.warn('Tenant::settings_json_schema is deprecated. Use AppConfiguration::Settings.json_schema_schema instead.')
-    AppConfiguration::Settings.json_schema
-  end
-
-  def self.style(*path)
-    ActiveSupport::Deprecation.warn('Tenant::style is deprecated. Use AppConfiguration::style instead.')
-    AppConfiguration.instance.style(*path)
-  end
-
-  def self.style_json_schema_str
-    ActiveSupport::Deprecation.warn('Tenant::style_json_schema_str is deprecated. Use AppConfiguration::style_json_schema_str instead.')
-    AppConfiguration.style_json_schema_str
-  end
-
-  def self.style_json_schema
-    ActiveSupport::Deprecation.warn('Tenant::style_json_schema is deprecated. Use AppConfiguration::style_json_schema instead.')
-    AppConfiguration.style_json_schema
-  end
-
-  def self.available_style_attributes
-    ActiveSupport::Deprecation.warn('Tenant::available_style_attributes is deprecated. Use AppConfiguration::available_style_attributes instead.')
-    AppConfiguration.available_style_attributes
+  def settings
+    ErrorReporter.report_msg('Tenant#settings is deprecated. Use AppConfiguration#settings instead.')
```

Comment:
No, we don't need it anymore. But since those two methods were used extensively throughout the codebase, I wanted to take some extra precautions with them. I felt like the deprecation warning wasn't fully effective because our logs are hard to search. So I took an extra step and reported any usage to Sentry (which is like raising an error without hurting the user). I'll set up a reminder for myself to remove those methods two weeks after the release if nothing gets reported.

Reply:
Alright, makes sense, let's just not forget that final step indeed